### ISSUE TITLE
Alpine support

### ISSUE THREAD

**Issue by bbaassssiiee** [robertdebock/ansible-role-bootstrap#50]
You commented out the alpine distro in `meta/main.yml`, any plans to add it back?

**Comment by robertdebock** [comment]
Let me see if Alpine works now. There was some idempotency issue.

```
    # Broken: idempotence, gather_facts: Failed to create temporary directory.
```

**Comment by robertdebock** [comment]
I've solved it locally, lets see if CI passes.

**Comment by bbaassssiiee** [comment]
TASK [bootstrap : wait for host] ***************************************************************************
skipping: [alpine]

TASK [bootstrap : test connection] *************************************************************************
fatal: [alpine]: FAILED! => {"changed": false, "elapsed": 3, "msg": "timed out waiting for ping module test: 'ping'"}

TASK [bootstrap : gather bootstrap facts] ******************************************************************
included: /Users/bas/code/ansible/ansiblebook/ansiblebook/ch18/playbooks/roles/bootstrap/tasks/gather_facts.yml for alpine

TASK [bootstrap : lookup bootstrap facts] ******************************************************************
fatal: [alpine]: UNREACHABLE! => {"changed": false, "msg": "Failed to connect to the host via ssh: root@127.0.0.1: Permission denied (publickey,password,keyboard-interactive).", "unreachable": true}

**Comment by bbaassssiiee** [comment]
I'd expect:

```yaml
  tasks:
    - name: apk update
      raw:  apk update

    - name: install python3
      raw:  apk add --update python3 py3-pip
```

**Comment by robertdebock** [comment]
Try again with version 5.2.7, released a moment ago. CI passes, so "works on my machine". ;-)

**Comment by bbaassssiiee** [comment]
`bootstrap_user: root` I cannot change that to work in all scenarios.

vagrant is a sudo-enabled `bootstrap_ansible_user` in my scenario's, package installs can be done as root.
Did you design for sudo snippets?

**Comment by bbaassssiiee** [comment]
TASK [bootstrap : install bootstrap packages] **********************************
task path: playbooks/roles/bootstrap/tasks/main.yml:60
failed: [rocky] (item=python3) => {"ansible_loop_var": "item", "changed": false, "item": "python3", "msg": "This command has to be run under the root user.", "results": []}
failed: [rocky] (item=sudo) => {"ansible_loop_var": "item", "changed": false, "item": "sudo", "msg": "This command has to be run under the root user.", "results": []}

**Comment by bbaassssiiee** [comment]
I could install on Alpine, but I made a PR to simplify user setup. I made PR to simplify the become issues.

**Comment by bbaassssiiee** [comment]
Test case https://github.com/ansiblebook/ansiblebook/tree/3rd-edition/ch18/playbooks

**Comment by robertdebock** [comment]
Ah, I was thinking what you meant with the _test case_, I think I understand;

Simpler code works, that's what the _test case_ proves.

But the simpler code is not idempotent. Some tasks are always `changed`. CI tests idempotency, I'm very much in favour of simpler code, but I have not found a simpler method for this role...

**Comment by bbaassssiiee** [comment]
If you have pointers on the configuration that I can use then please share/PR.

**Comment by robertdebock** [comment]
Fixed with version 6.0.0 release yesterday.
